experimental/list_formatter/src/deduplicating_array.rs

```diff
@@ -0,0 +1,111 @@
+use core::{mem, ops::Index, ptr};
+use icu_provider::yoke::*;
+
+#[derive(Debug)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Deserialize, serde::Serialize)
+)]
+// TODO: Make this generic over the 12. Serde currently can't derive
+// if const generics are present: https://github.com/serde-rs/serde/issues/1937
+pub struct DeduplicatingArray<T>([Dedupe<T>; 12]);
+
+#[derive(Debug, PartialEq)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Deserialize, serde::Serialize)
+)]
+#[serde(untagged)]
+enum Dedupe<T> {
+    Value(T),
```

Comment:
Thought: I don't know for sure how Postcard handles this, but when `T` is large, it means that an array of `[T; 12]` has to allocate space for 12 `T`s even if all but one of them is null.  Hopefully Postcard optimizes this, but I don't know. I suspect that there is some amount of indirection due to the use of an enum, which may explain the postcard size increase.

Reply:
Does `T` have a fixed size in postcard? It contains strings, are those offsets into a data section, or are the bytes encoded inline?
`Dedupe<T>` should have postcard-length 1 for the `Fallback` case, and the same length as `T` for the `Value` case due to the `untagged` setting.